test/gcs-test.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const crypto = require('bcrypto/lib/random');
+const assert = require('bsert');
+const GCSFilter = require('../lib/golomb/golomb');
+
+const key = crypto.randomBytes(16);
+const P = 20;
+
+const contents1 = [
+  Buffer.from('Alex', 'ascii'),
+  Buffer.from('Bob', 'ascii'),
+  Buffer.from('Charlie', 'ascii'),
+  Buffer.from('Dick', 'ascii'),
+  Buffer.from('Ed', 'ascii'),
+  Buffer.from('Frank', 'ascii'),
+  Buffer.from('George', 'ascii'),
+  Buffer.from('Harry', 'ascii'),
+  Buffer.from('Ilya', 'ascii'),
+  Buffer.from('John', 'ascii'),
+  Buffer.from('Kevin', 'ascii'),
+  Buffer.from('Larry', 'ascii'),
+  Buffer.from('Michael', 'ascii'),
+  Buffer.from('Nate', 'ascii'),
+  Buffer.from('Owen', 'ascii'),
+  Buffer.from('Paul', 'ascii'),
+  Buffer.from('Quentin', 'ascii')
+];
+
+const contents2 = [
+  Buffer.from('Alice', 'ascii'),
+  Buffer.from('Betty', 'ascii'),
+  Buffer.from('Charmaine', 'ascii'),
+  Buffer.from('Donna', 'ascii'),
+  Buffer.from('Edith', 'ascii'),
+  Buffer.from('Faina', 'ascii'),
+  Buffer.from('Georgia', 'ascii'),
+  Buffer.from('Hannah', 'ascii'),
+  Buffer.from('Ilsbeth', 'ascii'),
+  Buffer.from('Jennifer', 'ascii'),
+  Buffer.from('Kayla', 'ascii'),
+  Buffer.from('Lena', 'ascii'),
+  Buffer.from('Michelle', 'ascii'),
+  Buffer.from('Natalie', 'ascii'),
+  Buffer.from('Ophelia', 'ascii'),
+  Buffer.from('Peggy', 'ascii'),
+  Buffer.from('Queenie', 'ascii')
+];
+
+let filter1 = null;
+let filter2 = null;
+let filter3 = null;
+let filter4 = null;
+let filter5 = null;
+
+describe('GCS', function() {
```

Comment:
Let's initialize all filters by adding a `before()` before we start testing.
https://github.com/bcoin-org/bcoin/blob/f8118d0d9aef493b1ec93ccf026eaf863c9e7056/test/wallet-test.js#L245-L248

Reply:
Lets just move all the `let filter1...` lines inside the `describe` block to indicate the `it` blocks are linked. We are actually testing the construction methods, so they shouldnt go `before`